endoreg_db/serializers/label_video_segment/label_video_segment_update.py

```
@@ -0,0 +1,167 @@
+        # Key format: (start_frame_number, end_frame_number)
+        existing_segments_dict = {
+            (float(seg.start_frame_number), float(seg.end_frame_number)): seg
+            for seg in existing_segments
+        }
+
+        # Prepare lists for batch processing
+        # Initialize sets to track updates and new entries
+        updated_segments = set()
+        new_entries = []
+        existing_keys = set()
+        new_keys = set()
```

Ruff (0.12.2)

Local variable `new_keys` is assigned to but never used

Remove assignment to unused variable `new_keys`

(F841)